### ISSUE TITLE
Segfault by detaching pinned LFO

### ISSUE THREAD

**Issue by KottV** [BespokeSynth/BespokeSynth#51]
Hi,

Add any module with LFO (eg pulser).
Enable in-module LFO (RMB on slider), select pin.
Grab purple LFO's connection, press Backspace.

It affects Windows too, here the backtrace of the crash:

```
Thread 4 "Pool" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7ffff4f89700 (LWP 26642)]
FloatSliderLFOControl::GetLFOValue (this=this@entry=0x7fffbc7f6200, samplesIn=samplesIn@entry=0, forcePhase=<optimized out>) at ../../Source/FloatSliderLFOControl.cpp:256
256        return ofClamp(Interp(val, GetMin(), GetMax()), mTarget->GetMin(), mTarget->GetMax());
Missing separate debuginfos, use: zypper install krb5-debuginfo-1.18.2-3.2.x86_64 libX11-6-debuginfo-1.6.12-1.1.x86_64 libX11-xcb1-debuginfo-1.6.12-1.1.x86_64 libXau6-debuginfo-1.0.9-1.7.x86_64 libXcursor1-debuginfo-1.2.0-1.5.x86_64 libXext6-debuginfo-1.3.4-1.7.x86_64 libXfixes3-debuginfo-5.0.3-1.11.x86_64 libXinerama1-debuginfo-1.1.4-1.8.x86_64 libXrandr-devel-debuginfo-1.5.2-1.7.x86_64 libXrender1-debuginfo-0.9.10-1.12.x86_64 libXss1-debuginfo-1.2.3-1.9.x86_64 libbrotlicommon1-debuginfo-1.0.9-1.1.x86_64 libbrotlidec1-debuginfo-1.0.9-1.1.x86_64 libbz2-1-debuginfo-1.0.8-2.20.x86_64 libcom_err2-debuginfo-1.45.6-1.19.x86_64 libcrypt1-debuginfo-4.4.17-1.1.x86_64 libcurl4-debuginfo-7.72.0-1.2.x86_64 libdrm2-debuginfo-2.4.102-2.1.x86_64 libfreetype6-debuginfo-2.10.2-1.3.x86_64 libgcc_s1-debuginfo-10.2.1+git583-1.2.x86_64 libglvnd-debuginfo-1.3.2-2.1.x86_64 libidn2-0-debuginfo-2.3.0-3.2.x86_64 libkeyutils1-debuginfo-1.6-1.19.x86_64 libldap-2_4-2-debuginfo-2.4.53-57.1.x86_64 libnghttp2-14-debuginfo-1.41.0-1.3.x86_64 libopenssl1_1-debuginfo-1.1.1g-2.13.x86_64 libpcre1-debuginfo-8.44-1.19.x86_64 libpng16-16-debuginfo-1.6.37-1.7.x86_64 libpython3_8-1_0-debuginfo-3.8.5-2.1.x86_64 libsasl2-3-debuginfo-2.1.27-3.5.x86_64 libselinux1-debuginfo-3.0-2.3.x86_64 libssh4-debuginfo-0.9.4-1.4.x86_64 libstdc++6-debuginfo-10.2.1+git583-1.2.x86_64 libunistring2-debuginfo-0.9.10-2.8.x86_64 libxcb-dri3-0-debuginfo-1.14-1.3.x86_64 libxcb-glx0-debuginfo-1.14-1.3.x86_64 libxcb-present0-debuginfo-1.14-1.3.x86_64 libxcb-sync1-debuginfo-1.14-1.3.x86_64 libxcb1-debuginfo-1.14-1.3.x86_64 libz1-debuginfo-1.2.11-15.1.x86_64
(gdb) bt
#0  FloatSliderLFOControl::GetLFOValue (this=this@entry=0x7fffbc7f6200, samplesIn=samplesIn@entry=0, forcePhase=<optimized out>) at ../../Source/FloatSliderLFOControl.cpp:256
#1  0x000000000087e0d3 in FloatSliderLFOControl::DrawModule (this=0x7fffbc7f6200) at ../../Source/FloatSliderLFOControl.cpp:153
#2  0x000000000088410a in IDrawableModule::DrawFrame (this=0x7fffbc7f6200, w=106, h=213, drawModule=<optimized out>, titleBarHeight=@0x7ffff4f88ac8: 12, highlight=@0x7ffff4f88acc: 0) at ../../Source/IDrawableModule.cpp:304
#3  0x00000000008845f0 in IDrawableModule::Render (this=0x7fffbc7f6200) at ../../Source/IDrawableModule.cpp:355
#4  IDrawableModule::Render (this=0x7fffbc7f6200) at ../../Source/IDrawableModule.cpp:334
#5  0x00000000008a8983 in ModuleContainer::Draw (this=this@entry=0x144b848) at /usr/include/c++/10/bits/stl_vector.h:1043
#6  0x000000000089b067 in ModularSynth::Draw (this=this@entry=0x144b2f8, vg=<optimized out>) at ../../Source/ModularSynth.cpp:288
#7  0x000000000088e77b in MainContentComponent::render (this=0x144aca0) at ../../Source/MainComponent.cpp:303
#8  0x0000000000d031ca in juce::OpenGLAppComponent::renderOpenGL (this=<optimized out>) at /usr/share/juce/modules/juce_opengl/utils/juce_OpenGLAppComponent.cpp:62
#9  juce::OpenGLContext::CachedImage::renderFrame (this=this@entry=0x1448170) at /usr/share/juce/modules/juce_opengl/opengl/juce_OpenGLContext.cpp:262
#10 0x0000000000d03608 in juce::OpenGLContext::CachedImage::runJob (this=0x1448170) at /usr/share/juce/modules/juce_opengl/opengl/juce_OpenGLContext.cpp:482
#11 0x0000000000a62518 in juce::ThreadPool::runNextJob (this=0x14430f0, thread=...) at /usr/share/juce/modules/juce_core/threads/juce_ThreadPool.cpp:384
#12 0x0000000000a91eef in juce::ThreadPool::ThreadPoolThread::run (this=0x14a32e0) at /usr/share/juce/modules/juce_core/threads/juce_ThreadPool.cpp:36
#13 0x0000000000a5c45c in juce::Thread::threadEntryPoint (this=0x14a32e0) at /usr/share/juce/modules/juce_core/threads/juce_Thread.cpp:96
#14 0x0000000000a5c689 in juce::juce_threadEntryPoint (userData=<optimized out>) at /usr/share/juce/modules/juce_core/threads/juce_Thread.cpp:118
#15 juce::threadEntryProc (userData=<optimized out>) at /usr/share/juce/modules/juce_core/native/juce_posix_SharedCode.h:834
#16 0x00007ffff7b7aeaa in start_thread (arg=<optimized out>) at pthread_create.c:477
#17 0x00007ffff7362aff in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) list
251     float FloatSliderLFOControl::GetLFOValue(int samplesIn /*= 0*/, float forcePhase /*= -1*/)
252     {
253        float val = mLFO.Value(samplesIn, forcePhase);
254        if (mLFOSettings.mSpread > 0)
255           val = val * (1-mLFOSettings.mSpread) + (-cosf(val * FPI) + 1) * .5f * mLFOSettings.mSpread;
256        return ofClamp(Interp(val, GetMin(), GetMax()), mTarget->GetMin(), mTarget->GetMax());
257     }
258
259     void FloatSliderLFOControl::UpdateFromSettings()
260     {
(gdb)
```

**Comment by KottV** [comment]
And coredump from valgrind:

```
Reading symbols from ./BespokeSynth...
[New LWP 19202]
[New LWP 19523]
[New LWP 19203]
[New LWP 18418]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x000000000087de01 in FloatSlider::GetMax (this=0x0) at ../../Source/Slider.h:50
50         float GetMax() const { return mMax; }
[Current thread is 1 (Thread 0x6f22700 (LWP 19202))]
Missing separate debuginfos, use: zypper install krb5-debuginfo-1.18.2-3.2.x86_64 libX11-6-debuginfo-1.6.12-1.1.x86_64 libX11-xcb1-debuginfo-1.6.12-1.1.x86_64 libXau6-debuginfo-1.0.9-1.7.x86_64 libXcursor1-debuginfo-1.2.0-1.5.x86_64 libXext6-debuginfo-1.3.4-1.7.x86_64 libXfixes3-debuginfo-5.0.3-1.11.x86_64 libXinerama1-debuginfo-1.1.4-1.8.x86_64 libXrandr-devel-debuginfo-1.5.2-1.7.x86_64 libXrender1-debuginfo-0.9.10-1.12.x86_64 libXss1-debuginfo-1.2.3-1.9.x86_64 libbrotlicommon1-debuginfo-1.0.9-1.1.x86_64 libbrotlidec1-debuginfo-1.0.9-1.1.x86_64 libbz2-1-debuginfo-1.0.8-2.20.x86_64 libcom_err2-debuginfo-1.45.6-1.19.x86_64 libcrypt1-debuginfo-4.4.17-1.1.x86_64 libcurl4-debuginfo-7.72.0-1.2.x86_64 libdrm2-debuginfo-2.4.102-2.1.x86_64 libfreetype6-debuginfo-2.10.2-1.3.x86_64 libgcc_s1-debuginfo-10.2.1+git583-1.2.x86_64 libglvnd-debuginfo-1.3.2-2.1.x86_64 libidn2-0-debuginfo-2.3.0-3.2.x86_64 libkeyutils1-debuginfo-1.6-1.19.x86_64 libldap-2_4-2-debuginfo-2.4.53-57.1.x86_64 libnghttp2-14-debuginfo-1.41.0-1.3.x86_64 libopenssl1_1-debuginfo-1.1.1g-2.13.x86_64 libpcre1-debuginfo-8.44-1.19.x86_64 libpng16-16-debuginfo-1.6.37-1.7.x86_64 libpython3_8-1_0-debuginfo-3.8.5-2.1.x86_64 libsasl2-3-debuginfo-2.1.27-3.5.x86_64 libselinux1-debuginfo-3.0-2.3.x86_64 libssh4-debuginfo-0.9.4-1.4.x86_64 libstdc++6-debuginfo-10.2.1+git583-1.2.x86_64 libunistring2-debuginfo-0.9.10-2.8.x86_64 libxcb-dri3-0-debuginfo-1.14-1.3.x86_64 libxcb-glx0-debuginfo-1.14-1.3.x86_64 libxcb-present0-debuginfo-1.14-1.3.x86_64 libxcb-sync1-debuginfo-1.14-1.3.x86_64 libxcb1-debuginfo-1.14-1.3.x86_64 libz1-debuginfo-1.2.11-15.1.x86_64 valgrind-debuginfo-3.16.1-1.2.x86_64
(gdb) bt
#0  0x000000000087de01 in FloatSlider::GetMax (this=0x0) at ../../Source/Slider.h:50
#1  FloatSliderLFOControl::GetLFOValue (this=this@entry=0x301a9bf0, samplesIn=samplesIn@entry=0, forcePhase=<optimized out>) at ../../Source/FloatSliderLFOControl.cpp:256
#2  0x000000000087e0d3 in FloatSliderLFOControl::DrawModule (this=0x301a9bf0) at ../../Source/FloatSliderLFOControl.cpp:153
#3  0x000000000088410a in IDrawableModule::DrawFrame (this=0x301a9bf0, w=106, h=213, drawModule=<optimized out>, titleBarHeight=@0x6f21ac8: 12, highlight=@0x6f21acc: 0) at ../../Source/IDrawableModule.cpp:304
#4  0x00000000008845f0 in IDrawableModule::Render (this=0x301a9bf0) at ../../Source/IDrawableModule.cpp:355
#5  IDrawableModule::Render (this=0x301a9bf0) at ../../Source/IDrawableModule.cpp:334
#6  0x00000000008a8983 in ModuleContainer::Draw (this=this@entry=0x62351a8) at /usr/include/c++/10/bits/stl_vector.h:1043
#7  0x000000000089b067 in ModularSynth::Draw (this=this@entry=0x6234c58, vg=<optimized out>) at ../../Source/ModularSynth.cpp:288
#8  0x000000000088e77b in MainContentComponent::render (this=0x6234600) at ../../Source/MainComponent.cpp:303
#9  0x0000000000d031ca in juce::OpenGLAppComponent::renderOpenGL (this=<optimized out>) at /usr/share/juce/modules/juce_opengl/utils/juce_OpenGLAppComponent.cpp:62
#10 juce::OpenGLContext::CachedImage::renderFrame (this=this@entry=0x2d7b3020) at /usr/share/juce/modules/juce_opengl/opengl/juce_OpenGLContext.cpp:262
#11 0x0000000000d03608 in juce::OpenGLContext::CachedImage::runJob (this=0x2d7b3020) at /usr/share/juce/modules/juce_opengl/opengl/juce_OpenGLContext.cpp:482
#12 0x0000000000a62518 in juce::ThreadPool::runNextJob (this=0x2d6fbbc0, thread=...) at /usr/share/juce/modules/juce_core/threads/juce_ThreadPool.cpp:384
#13 0x0000000000a91eef in juce::ThreadPool::ThreadPoolThread::run (this=0x6296230) at /usr/share/juce/modules/juce_core/threads/juce_ThreadPool.cpp:36
#14 0x0000000000a5c45c in juce::Thread::threadEntryPoint (this=0x6296230) at /usr/share/juce/modules/juce_core/threads/juce_Thread.cpp:96
#15 0x0000000000a5c689 in juce::juce_threadEntryPoint (userData=<optimized out>) at /usr/share/juce/modules/juce_core/threads/juce_Thread.cpp:118
#16 juce::threadEntryProc (userData=<optimized out>) at /usr/share/juce/modules/juce_core/native/juce_posix_SharedCode.h:834
#17 0x0000000004c84eaa in start_thread (arg=<optimized out>) at pthread_create.c:477
#18 0x00000000054e7aff in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
(gdb) list
45         FloatSliderLFOControl* AcquireLFO();
46         void MatchExtents(FloatSlider* slider);
47         void SetRelative(bool relative) { mRelative = relative; }
48         void SetClamped(bool clamped) { mClamped = clamped; }
49         float GetMin() const { return mMin; }
50         float GetMax() const { return mMax; }
51         void SetMaxValueDisplay(string display) { mMaxValueDisplay = display; }
52         void SetMinValueDisplay(string display) { mMinValueDisplay = display; }
53         void SetLFO(FloatSliderLFOControl* lfo);
54         void SetShowName(bool show) { mShowName = show; }
(gdb)
```

**Comment by awwbees** [comment]
fixed in 97e36a9b79fcac22ba6ee77a9b0e6b3c2e90d805

**Comment by KottV** [comment]
Thank You!
